preprocessing/shift_generation.py

```diff
@@ -217,9 +239,18 @@ def get_t_covered_by_off_shifts(off_shifts, time_sets):
     time_periods = time_sets["periods"][0]
 
     for shift in off_shifts:
-        end = time_periods.index(shift[0] + shift[1])
-        start = time_periods.index(shift[0])
-        t_covered[shift[0], shift[1]] = time_periods[start:end]
+        try:
+            end = time_periods.index(shift[0] + shift[1])
```

Comment:
Why not `end = shift[0] + shift[1]`?

Reply:
Because it is the index in time_periods you want, not the time.